Sources/Build/ManifestBuilder.swift

```diff
@@ -218,22 +219,35 @@ extension LLBuildManifestBuilder {
                 let jobInputs = job.inputs.map { $0.resolveToNode() }
                 let jobOutputs = job.outputs.map { $0.resolveToNode() }
 
+                // Add target dependencies as inputs to the main module build command.
+                //
+                // Jobs for a target's intermediate build artifacts, such as PCMs or
+                // modules built from a .swiftinterface, do not have a
+                // dependency on cross-target build products. If multiple targets share
+                // common intermediate dependency modules, such dependencies can lead
+                // to cycles in the resulting manifest.
+                var manifestNodeInputs = jobInputs
+                if !job.inputs.contains(where: { $0.type == .swiftInterface }) &&
```

Comment:
Should this predicate be given a name and sunk into swift-driver itself?

Reply:
Yeah, that's a good idea.